### PR TITLE
[FEAT] I18n language selection UI and store locale

### PR DESCRIPTION
**Pull request by hmqgg**
`client/src/core/components/languageDropdown.vue` implements a dropdown selection for language switching.
It hard-coded supported languages and I do think it's acceptable for the current use case since the translation file is webpacked.
Maybe it needs a better UI/UX, but I'm not good at this.

After language selected, it will call localStorage API and stores the locale info for next time loading such as refreshing.

### CODE REVIEW

File: client/src/auth/login.vue [hunk 1]
```diff
@@ -100,6 +106,11 @@ export default class Login extends Vue {
                     </span>
                 </div>
 
+                <div class="input">
+                    <label for="langDropdown" v-t="'locale.select'"></label>
+                    <languageDropdown id="langDropdown" />
```
Comment (Kruptein) [inline client/src/auth/login.vue:111]: There is no need to specify an `id` if you never use it in css/js

File: client/src/core/components/languageDropdown.vue [hunk 2]
```diff
@@ -0,0 +1,34 @@
+<script lang="ts">
+import Vue from "vue";
+import Component from "vue-class-component";
+
+import { Prop } from "vue-property-decorator";
+
+@Component
+export default class LanguageDropdown extends Vue {
+    @Prop() value!: string;
```
Comment (Kruptein) [inline client/src/core/components/languageDropdown.vue:9]: This seems to be unused?

File: client/src/i18n.ts [hunk 3]
```diff
@@ -17,7 +17,7 @@ function loadLocaleMessages(): LocaleMessages {
 }
 
 export default new VueI18n({
-    locale: process.env.VUE_APP_I18N_LOCALE || "en",
+    locale: localStorage.getItem("locale") || process.env.VUE_APP_I18N_FALLBACK_LOCALE || "en",
```
Comment (Kruptein) [inline client/src/i18n.ts:20]: Why did this become fallback locale ?

```suggestion
    locale: localStorage.getItem("locale") || process.env.VUE_APP_I18N_LOCALE || "en",
```

File: client/src/locales/de.json [hunk 4]
```diff
@@ -1,4 +1,10 @@
 {
+    "locale": {
+        "select": "Select Language:",
+        "en": "English",
+        "zh": "中文",
+        "de": "Deustch"
```
Comment (Kruptein) [inline client/src/locales/de.json:6]:
```suggestion
        "de": "Deutsch"
```

File: client/src/locales/en.json [hunk 5]
```diff
@@ -1,4 +1,10 @@
 {
+    "locale": {
+        "select": "Select Language:",
+        "en": "English",
+        "zh": "中文",
+        "de": "Deustch"
```
Comment (Kruptein) [inline client/src/locales/en.json:6]:
```suggestion
        "de": "Deutsch"
```

File: client/src/locales/de.json (hunk 4 above)
Comment (hmqgg) [reply client/src/locales/de.json:6]: typo

File: client/src/core/components/languageDropdown.vue (hunk 2 above)
Comment (hmqgg) [reply client/src/core/components/languageDropdown.vue]: Removed

**Pull request merged**